src/foam/nanos/notification/email/EmailTemplate.js

```diff
@@ -54,16 +77,133 @@ bodyAsByteArrayIsSet_ = false;`
       transient: true,
       type: 'Byte[]',
       javaFactory: 'return getBody() != null ? getBody().getBytes(StandardCharsets.UTF_8) : null;'
-    },
+    }
+  ],
+  methods: [
     {
-      class: 'String',
-      name: 'displayName'
+      name: 'apply',
+      type: 'foam.nanos.notification.email.EmailMessage',
+      args: [
+        {
+          name: 'x',
+          type: 'Context',
+        },
+        {
+          name: 'group',
+          class: 'String',
+          documentation: 'group of user whose the recipient of the email being sent'
+        },
+        {
+          name: 'emailMessage',
+          type: 'foam.nanos.notification.email.EmailMessage',
+          documentation: 'Email message'
+        },
+        {
+          name: 'templateArgs',
+          type: 'Map',
+          documentation: 'Template arguments'
+        }
+      ],
+      javaCode: `
+        Logger logger = (Logger) x.get("logger");
+        
+        if ( emailMessage == null ) {
+          logger.error("emailMessage is Null");
+          return null;
+        }
+
+        String tempKeyString = "";
+        Object value = null;
+        JtwigModel model = null;
+        EnvironmentConfiguration config = EnvironmentConfigurationBuilder
+          .configuration()
+            .resources()
+              .resourceLoaders()
+                .add(new TypedResourceLoader("dao", new DAOResourceLoader(x, group)))
+                .and()
+            .and()
+          .build();
+
+        for ( Object key : templateArgs.keySet() ) {
+          value = templateArgs.get((String)key);
+          if ( value instanceof String ) {
+            tempKeyString = (String) value;
+            templateArgs.put((String) key, new String(tempKeyString.getBytes(StandardCharsets.ISO_8859_1), StandardCharsets.UTF_8));
+          }
+          model = JtwigModel.newModel(templateArgs);
+          if ( model == null ) {
+            logger.error("JtwigModel is Null");
+            return null;
+          }
+        }
+
+        return fillInEmailProperties_(x, emailMessage, model, config);
+      `
     },
     {
-      class: 'String',
-      name: 'sendTo',
-      documentation: `This property will set to whomever the email is being sent
-        to.`
+      name: 'fillInEmailProperties_',
+      type: 'foam.nanos.notification.email.EmailMessage',
+      documentation: 'Applies template properties to emailMessage, where emailMessage property is empty',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        },
+        {
+          name: 'emailMessage',
+          type: 'foam.nanos.notification.email.EmailMessage'
+        },
+        {
+          name: 'model',
+          javaType: 'org.jtwig.JtwigModel'
+        },
+        {
+          name: 'config',
+          javaType: 'EnvironmentConfiguration'
+        }
+      ],
+      javaCode: `
+        // BODY:
+        if ( SafetyUtil.isEmpty(emailMessage.getBody()) ) 
+        {
```

Comment:
This isn't really our style for java. If you want a formatter to come into play, open the generated java file, run a formatter on it, find the method you implemented, and copy/paste it back here.